presto-main/src/main/java/com/facebook/presto/util/DateTimeUtils.java

```diff
@@ -134,6 +137,33 @@ public static long parseTimestampWithoutTimeZone(TimeZoneKey timeZoneKey, String
         return TIMESTAMP_WITH_OR_WITHOUT_TIME_ZONE_FORMATTER.withChronology(getChronology(timeZoneKey)).parseMillis(value);
     }
 
+    public static long timestampWithTimeZoneToDate(long value)
+    {
+        // round down the current timestamp to days
+        ISOChronology chronology = unpackChronology(value);
+        long date = chronology.dayOfYear().roundFloor(unpackMillisUtc(value));
+        // date is currently midnight in timezone of the original value
+        // we expect midnight in UTC so we shift to that timezone
+        long millis = date + chronology.getZone().getOffset(date);
+        return TimeUnit.MILLISECONDS.toDays(millis);
+    }
+
+    public static long timestampWithTimeZoneToTimestampWithoutTimeZone(long value)
+    {
+        return unpackMillisUtc(value);
```

Comment:
This code is only moved from operators inside TimestampWithTimeZoneOperators.java in order to be reused. Those are converters between different encoding of DateTime types on long variable.